rpcs3/Emu/Cell/Modules/cellSysutil.cpp

```diff
@@ -283,18 +283,29 @@ s32 cellSysutilUnregisterCallback(u32 slot)
 	return CELL_OK;
 }
 
-s32 cellSysCacheClear(void)
+s32 cellSysCacheClear()
 {
-	cellSysutil.todo("cellSysCacheClear()");
+	// Checks if the param exists, else we quit
 
 	if (!fxm::check<CellSysCacheParam>())
 	{
 		return CELL_SYSCACHE_ERROR_NOTMOUNTED;
 	}
 
-	const std::string& local_path = vfs::get("/dev_hdd1/cache/");
+	// Get the param as a shared ptr, then decipher the cacheid from it
+	// (Instead of assuming naively that the param is passed as argument)
+	std::shared_ptr<CellSysCacheParam> param = fxm::get<CellSysCacheParam>();
+	const std::string& cache_id = param->cacheId;
```

Comment:
you still need to check for
`if (!param) return CELL_SYSCACHE_ERROR_ACCESS_ERROR;`
or something like that

Reply:
But I told you to...
Also - you _can_ change existing code you know :p